logstash-core-event-java/src/main/java/com/logstash/Javafier.java

```diff
@@ -127,6 +164,9 @@ public static Object deep(IRubyObject o) {
             case RubyNil: return deep((RubyNil)o);
             case True: return deep((RubyBoolean.True)o);
             case False: return deep((RubyBoolean.False)o);
+            case MapJavaProxy: return deep(((MapJavaProxy) o).to_hash());
```

Comment:
does the `.to_hash()` call involve any copying? did not dig into this but we should make sure we are not ending up doing a double-conversion here